### PR TITLE
Implement api search book

### PR DESCRIPTION
**Pull request by mvn-minhphan-dn**
## Description
Implement api search book.
## Reference Requirements
- Product Backlog ID: PBI-33
- Task ID: TI-2

## Reference Isuees
None

## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

- .gitignore
- app/Http/Controllers/Api/BookController.php
- app/Helpers/common.php
- resources/views/api-doc-builders/index.blade.php
- tests/Feature/BookApiTest.php

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- select `id`, `name`, `author`, `avg_rating`, `total_rating`, (select count(*) from `borrowings` where `books`.`id` = `borrowings`.`book_id` and `borrowings`.`deleted_at` is null) as `borrowings_count` from `books` where `books`.`deleted_at` is null and `books`.`name` like '%ABC%' or `books`.`author` like '%ABC%' order by `id` desc limit 10 offset 0

image

### CODE REVIEW

File: app/Http/Controllers/Api/BookController.php [hunk 1]
```diff
@@ -117,20 +117,23 @@ public function restore($id)
...
             'avg_rating'
         ];
         $books = Book::select($fields)
-                    ->orderBy('created_at', 'DESC')
-                    ->paginate(config('define.book.item_limit'));
+            ->where('name', 'like', "%$request->search%")
+            ->orWhere('author', 'like', "%$request->search%")
+            ->orderBy('created_at', 'desc')
+            ->paginate(config('define.book.item_limit'));
         $meta = [
```
Comment (at-phutran2) [inline app/Http/Controllers/Api/BookController.php]: return metaResponse($books, Response::HTTP_OK);

**Review by at-phutran2 — COMMENTED** [review]
LGTM!

File: app/Http/Controllers/Api/BookController.php (hunk 1 above)
Comment (at-tuanphan) [reply app/Http/Controllers/Api/BookController.php]: common được merged rồi em pull về rồi update giúp anh nha

**Review by ghost — COMMENTED** [review]
please check my comment

File: app/Http/Controllers/Api/BookController.php (hunk 1 above)
Comment (at-minhphan) [reply app/Http/Controllers/Api/BookController.php]: dạ vâng ạ

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**